capstone/src/main/java/com/google/step/clients/ScoreClient.java

```diff
@@ -0,0 +1,164 @@
+package com.google.step.clients;
+
+import com.google.step.data.RestaurantPageViews;
+import com.google.step.data.RestaurantScore;
+import com.google.step.data.WeeklyPageView;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Collections;
+
+
+/**
+ * Will take care of calculating the score for each restaurant.
+ */
+public class ScoreClient {
+
+    private final double minimumWeeks = 10.0;
+    private final MetricsClient metricsClient = new MetricsClient();
+
+    /**
+     * Calculates the scores of each restaurant.
+     * @return an unmodifiable list of restaurantScore class.
+     */
+    public List<RestaurantScore> calculateScores() {
+        Map<Long, Double> scoreMap = new HashMap<>();
+        List<Long> restaurantIds = new ArrayList<>();
+
+        List<RestaurantPageViews> allPageViews = metricsClient.getAllPageViews();
+
+        double systemAverage = getSystemAverage(allPageViews);
+
+        // Calculate score
+        for (RestaurantPageViews restaurant: allPageViews) {
+            double restaurantAverage = calculateRestaurantAveragePageViews(restaurant);
+
+            // Since the data is sorted we can just get the latest pageView for that restaurant.
+            List<WeeklyPageView> pageViewList = restaurant.getPageViews();
+            int lastPageViewIndex = pageViewList.size() - 1;
+            int latestPageView = pageViewList.get(lastPageViewIndex).getCount();
+
+            double score = calculateRawScore(latestPageView, restaurantAverage, systemAverage);
+            restaurantIds.add(Long.parseLong(restaurant.getId()));
```

Comment:
Does the `RestaurantPageViews` class have a `getId()` method?

Reply:
I believe this field was added in his most recent iteration.

Reply:
Yup!